pkg/querier/batch/stream.go

```diff
@@ -61,7 +47,7 @@ func (bs *batchStream) at() (int64, float64) {
 	return b.Timestamps[b.Index], b.Values[b.Index]
 }
 
-// mergeBatches assumes the contents of batches are overlapping and unstorted.
+// mergeBatches assumes the contents of batches are overlapping and unsorted.
 // Merge them together into a sorted, non-overlapping stream in result.
 // Caller must guarantee result is big enough.  Return value will always be a
```

Comment:
Can this comment be relied on?  Seems the code would be a lot simpler if it were.